src/serializer/ResidualHeapSerializer.js

```diff
@@ -574,52 +621,11 @@ export class ResidualHeapSerializer {
         );
       });
       remainingProperties.delete("length");
-    } else {
-      // An array's length property cannot be redefined, so this won't run user code
-      let len = ToLength(realm, lenProperty);
-      for (let i = 0; i < len; i++) {
-        let key = i + "";
-        let propertyBinding = remainingProperties.get(key);
-        let elem = null;
```

Comment:
Thanks. This is a bit tricky: if I use "delete a[2]" it won't leave hole in the middle of array, while if I used "var a = [1, 2,,,5]" it will. My new commit handles it.